Wordle.Api/Wordle.Api/Services/PlayerService.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.EntityFrameworkCore;
+using Wordle.Api.Data;
+using Wordle.Api.Dtos;
+
+namespace Wordle.Api.Services
+{
+    public class PlayerService
+    {
+        
+        private readonly AppDbContext _db;
+
+        public PlayerService(AppDbContext db)
+        {
+            _db = db;
+        }
+
+        public async Task<IEnumerable<Player>> GetTopTen()
+        {
+
+            var topTen = await _db.Players
+                .OrderBy(p => p.AverageAttempts)
+                .ThenByDescending(p => p.GameCount)
+                .Take(10)
+                .ToListAsync();
+            return topTen;
+
+        }
+        
+        public async Task<Player> AddPlayer(string Name, int TotalSecondsPlayed, int TotalAttempts)
+        {
+            if(Name == null) { throw new ArgumentException("Name can't be null"); }
+            var player = await _db.Players.FirstOrDefaultAsync(p =>  p.Name == Name);
+            if(player != null) 
```

Comment:
Instead of having an if/else here, you could actually combine your update & new into one step by just checking if player is null from line 32 and then setting it to a new player if it is... And then just update/set the properties all in one step.